src/Orleans.sln

```diff
@@ -271,4 +271,7 @@ Global
 	GlobalSection(NestedProjects) = preSolution
 		{0BB483EF-6401-424C-9C4F-5750B300C7FD} = {DD5D35C0-A5FE-41CC-A10D-CAAA37996479}
 	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		EnterpriseLibraryConfigurationToolBinariesPathV6 = packages\EnterpriseLibrary.TransientFaultHandling.6.0.1304.0\lib\portable-net45+win+wp8
```

Comment:
Just leaving my 2c here...

@jthelin I agree with you that Nuget on .csproj is adding 4 digit references on the project file. 

The ‘guilty’ there is not on Nuget. VS uses the AssemblyInfo.cs to add the assembly information there and it is on 0.0.0.0 format while Nuget uses Semantic version which is 0.0.0. The problem is with VS which uses the 4-digit format.

If you have a look on any coreCLR project, they fixed it and even .Net Framework libraries/references are now following Semantic Version.

> By arbitrarily elevating scoped dependencies from one project where they are needed to the whole solution where they are not. Q.E.D. ;)

The scoped dependency is not something Nuget does automatically. The developer of the package has requested to modify the solution file the same way we do when add MSBuild tasks on Orleans-based projects to use compile-time codegen. We can’t judge that.